spirv_glsl.cpp

```diff
@@ -1017,6 +1017,27 @@ void CompilerGLSL::emit_uniform(const SPIRVariable &var)
 	statement(layout_for_variable(var), "uniform ", variable_decl(var), ";");
 }
 
+void CompilerGLSL::replace_illegal_names()
+{
+	for (auto &id : ids)
+	{
+		if (id.get_type() == TypeVariable)
+		{
+			auto &var = id.get<SPIRVariable>();
+			auto &type = get<SPIRType>(var.basetype);
+
+			if (!is_builtin_variable(var))
```

Comment:
You also need to check for var.remapped_variable. Remapped variables are not declared, and they must be able to have "reserved" GL names, because they aren't declared.